File: app/src/main/java/ru/ls/donkitchen/app/DonKitchenApplication.java
package ru.ls.donkitchen.app;

import android.app.Application;

import com.crashlytics.android.Crashlytics;
import com.facebook.stetho.Stetho;
import com.facebook.stetho.okhttp.StethoInterceptor;
import com.squareup.okhttp.OkHttpClient;

import io.fabric.sdk.android.Fabric;

/**
 * Главный класс приложения
 *
 * @author Lord (Kuleshov M.V.)
 * @since 26.08.15
 */
public class DonKitchenApplication extends Application {
	private AppComponent applicationComponent;

	@Override
	public void onCreate() {
		super.onCreate();

		Stetho.initializeWithDefaults(this);

		OkHttpClient client = new OkHttpClient();
		client.networkInterceptors().add(new StethoInterceptor());

		Fabric.with(this, new Crashlytics());

		applicationComponent = DaggerAppComponent.builder()
				.appModule(new AppModule(this))
				.build();
		applicationComponent.inject(this);
	}

	public AppComponent component() {
		return applicationComponent;
	}
}

File: app/src/main/kotlin/ru/ls/donkitchen/app/DonKitchenApplication.kt
package ru.ls.donkitchen.app

import androidx.multidex.MultiDexApplication

/**
 * Главный класс приложения
 */
open class DonKitchenApplication : MultiDexApplication() {
    private lateinit var component: AppComponent

    override fun onCreate() {
        super.onCreate()

        instance = this
        createComponent()
        component.inject(this)
    }

    private fun createComponent(): AppComponent {
        component = DaggerAppComponent.builder().appModule(AppModule(this)).build()

        return component
    }

    fun component(): AppComponent {
        return component
    }

    companion object {
        private lateinit var instance: DonKitchenApplication

        @JvmStatic
        fun instance(): DonKitchenApplication {
            return instance
        }
    }
}
